src/app/components/SelectNode/BTCPayServer.tsx

```diff
@@ -12,80 +16,101 @@ export interface BTCPayServerConfig {
   readonlyMacaroon?: string;
 }
 
-interface Props {
+interface StateProps {
+  nodeInfo: AppState['node']['nodeInfo'];
+  isCheckingAuth: AppState['node']['isCheckingAuth'];
+  checkAuthError: AppState['node']['checkAuthError'];
+}
+
+interface DispatchProps {
+  checkAuth: typeof checkAuth;
+}
+
+interface OwnProps {
   error?: Error | null;
-  submitConfig(config: BTCPayServerConfig): void;
 }
 
+type Props = StateProps & DispatchProps & OwnProps & RouteComponentProps;
+
 interface State {
   configData: string;
   isSubmitting: boolean;
 }
 
-export default class BTCPayServer extends React.Component<Props, State> {
+class BTCPayServer extends React.Component<Props, State> {
   state: State = {
     configData: '',
-    isSubmitting: false,
+    isSubmitting: false
   };
 
   componentWillUpdate(nextProps: Props) {
+    if (this.props.nodeInfo) {
+      this.props.history.push('/onboarding-node-confirm');
+    }
     if (nextProps.error && this.props.error !== nextProps.error) {
       message.error(nextProps.error.message);
     }
   }
 
   render() {
-    const { configData, isSubmitting } = this.state;
+    const {checkAuthError} = this.props;
+    const {configData, isSubmitting} = this.state;
     const url = this.getConfigDataUrl(configData);
     const error = configData && !url ? 'Invalid config data' : undefined;
 
     return (
-      <div className="BTCPayServer">
-        <p>
-          Follow these steps to connect your BTCPay Server to Joule. Your node
-          must be fully synced in order to get connection details.
-        </p>
-        <ol className="BTCPayServer-steps">
-          <li>Navigate to your BTCPayServer and log in as an admin</li>
-          <li>Go to Server Settings > Services > LND Rest - See information</li>
-          <li>Click "See QR Code information" and copy the QR Code data</li>
-          <li>Paste the data below:</li>
-        </ol>
-        <Form className="BTCPayServer-form" onSubmit={this.handleSubmit} layout="vertical">
-          <Form.Item
-            label="Connect data"
-            help={error}
-            validateStatus={error ? 'error' : url ? 'success' : undefined}
-          >
-            <Input
+      <div>
+        <h2 className="SelectNode-title">Connect to your BTCPay Server</h2>
```

Comment:
It might just be better to make a `<SelectNodeTemplate title="Title Goes Here">` component than having to implicitly copy the styles and classnames across 5 components. That way future changes (e.g. back button) would only require changes in one place.

Reply:
True that. Good call.

Reply:
This is handled in the latest commit by using a `<TitleTemplate>` instead of `<h2>` elements.